domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -884,63 +884,66 @@ public void saveBatch(String table, JsonArray entities, Handler<AsyncResult<Resu
       return;
     }
-    client.getConnection(res -> {
-      if (res.failed()) {
-        replyHandler.handle(Future.failedFuture(res.cause()));
+    if (log.isInfoEnabled()) {
+      log.info("starting: saveBatch size=" + entities.size());
+    }
+    StringBuilder sql = new StringBuilder()
+      .append(INSERT_CLAUSE)
+      .append(schemaName).append(DOT).append(table)
+      .append(" (").append(column).append(") VALUES (?)");
+    for (int i = 1; i < entities.size(); i++) {
+      sql.append(",(?)");
+    }
+    sql.append(" RETURNING ").append(idField);
+
+    SQLConnection connection = sqlConnection.result();
+    connection.queryWithParams(sql.toString(), entities, queryRes -> {
+      if (queryRes.failed()) {
+        log.error("saveBatch size=" + entities.size()
+          + SPACE
+          + queryRes.cause().getMessage(),
+          queryRes.cause());
+        statsTracker("saveBatchFailed", table, start);
+        replyHandler.handle(Future.failedFuture(queryRes.cause()));
         return;
       }
-
       if (log.isInfoEnabled()) {
-        log.info("starting: saveBatch size=" + entities.size());
-      }
-      SQLConnection connection = res.result();
-      StringBuilder sql = new StringBuilder()
-          .append(INSERT_CLAUSE)
-          .append(schemaName).append(DOT).append(table)
-          .append(" (").append(column).append(") VALUES (?)");
-      for (int i=1; i<entities.size(); i++) {
-        sql.append(",(?)");
+        log.info("success: saveBatch size=" + entities.size());
       }
-      sql.append(" RETURNING ").append(idField);
-
-      connection.queryWithParams(sql.toString(), entities, queryRes -> {
-        connection.close();
-        if (queryRes.failed()) {
-          log.error("saveBatch size=" + entities.size()
-            + SPACE +
-              queryRes.cause().getMessage(),
-              queryRes.cause());
-          statsTracker("saveBatchFailed", table, start);
-          replyHandler.handle(Future.failedFuture(queryRes.cause()));
-          return;
-        }
-        if (log.isInfoEnabled()) {
-          log.info("success: saveBatch size=" + entities.size());
-        }
-        statsTracker("saveBatch", table, start);
-        replyHandler.handle(Future.succeededFuture(queryRes.result()));
-      });
+      statsTracker("saveBatch", table, start);
+      replyHandler.handle(Future.succeededFuture(queryRes.result()));
     });
   }
 
-  /***
-   * Save a list of POJOs.
-   * POJOs are converted to json and saved in a single sql call. The generated IDs of the
-   * inserted records are returned in the ResultSet.
-   * @param table  destination table to insert into
-   * @param entities  each list element is a POJO
-   * @param replyHandler  result, containing the id field for each inserted POJO
-   */
-  public void saveBatch(String table, List<Object> entities, Handler<AsyncResult<ResultSet>> replyHandler) {
+  private JsonArray batchEntitiesToJsonArray(List<Object> entities,
```

Comment:
There is no docum,entation when batchEntitiesToJsonArray invokes the replyHandler.
I don't think that using the replyHandler in this method is a good idea. Simply throw the exception and let the caller handle it, that is more easy to understand and the Exception must be created anyhow.